rllib/evaluation/collectors/agent_collector.py

```diff
@@ -116,6 +116,15 @@ def __init__(
         # The simple timestep count for this agent. Gets increased by one
         # each time a (non-initial!) observation is added.
         self.agent_steps = 0
+        # Keep track of view requirements that have a view on columns that we gain from
+        # inference and also need for inference. These have dummy values prepended in
+        # buffers to account for the missing value when building for inference
+        # Example: We have one 'state_in' view requirement that has a view on our
+        # state_outs at t=[-10, ..., -1]. At any given build_for_inference()-call,
+        # the buffer must contain eleven values from t=[-10, ..., 0] for us to index
+        # properly. Since state_out at t=0 is missing, we substitute it with a buffer
+        # value that should never make it into batches built for training.
+        self.gained_from_inference = set()
```

Comment:
Suggestion: I personally will forget what `gained_from_inference` actually means in the future. I think a better name would be something like `data_cols_with_dummy_values`. We also need to use this opportunity, while we have fresh memory about the details, to update the main docstring of the agent_collector class to explain the content of the buffer and how the dummy value will be treated during both inference and training. I'll write a draft when the other matters are resolved.

Reply:
As per our discussion: Changed!